yarn-project/foundation/src/eth-address/index.ts

```diff
@@ -207,14 +207,23 @@ export class EthAddress {
     return Fr.fromBuffer(this.toBuffer32());
   }
 
+  /**
+   * Converts a field to a eth address.
+   * @param fr - The field to convert.
+   * @returns The eth address.
+   */
+  static fromField(fr: Fr): EthAddress {
+    return new this(fr.toBuffer().slice(-EthAddress.SIZE_IN_BYTES));
+  }
+
   /**
    * Deserializes from a buffer or reader, corresponding to a write in cpp.
    * @param buffer - Buffer to read from.
    * @returns The EthAdress.
    */
   static fromBuffer(buffer: Buffer | BufferReader): EthAddress {
     const reader = BufferReader.asReader(buffer);
-    return new EthAddress(reader.readBuffer());
+    return new this(reader.readBuffer());
```

Comment:
```suggestion
    return new EthAddress(reader.readBuffer());
```

Using the explicit class name here prevents hiccups with `this`. Eg if you use `this`, you cannot do something like `myArrayofBuffers.map(EthAddress.fromBuffer)`.